mm2src/mm2_main/src/lp_stats.rs

```diff
@@ -170,16 +170,20 @@ struct Mm2VersionRes {
     nodes: HashMap<String, String>,
 }
 
-fn process_get_version_request(ctx: MmArc) -> Result<Option<Vec<u8>>, String> {
+fn process_get_version_request(ctx: MmArc) -> Result<Vec<u8>, String> {
     let response = ctx.mm_version().to_string();
-    let encoded = try_s!(encode_message(&response));
-    Ok(Some(encoded))
+    encode_message(&response).map_err(|e| e.to_string())
 }
 
-pub fn process_info_request(ctx: MmArc, request: NetworkInfoRequest) -> Result<Option<Vec<u8>>, String> {
-    log::debug!("Got stats request {:?}", request);
+fn process_get_peer_utc_timestamp_request() -> Result<Vec<u8>, String> {
+    let timestamp = common::get_utc_timestamp();
+    encode_message(&timestamp).map_err(|e| e.to_string())
+}
```

Comment:
...encodes it as a `i64`. we should stick to only one. `u64` makes the most sense application-wise as we don't really need the negative side? right?

but curious, is the encoding of `i64` and `u64` compatible?

Reply:
> we should stick to only one. u64 makes the most sense application-wise as we don't really need the negative side? right?

Why do you want to cast the value again when you can simply encode it?

> but curious, is the encoding of i64 and u64 compatible?

They are

Reply:
> Why do you want to cast the value again when you can simply encode it?

didn't know the encoding for both is compatible, still not sure how the negative side is encoded (probably as rust's then, will checkout).
so thing is, you actually do some kind of an implicit cast here using data encoding and decoding, you encode the number as `i64`, send it, decode it as `u64` from the receiver. i assume a negative `i64` will decode fine as a `u64` just like what the `as` keyword does but give a garbage data (will check more on that encoding lib).

what i am suggesting is, we can encode it as `i64` and decode it as `i64` as well, we check `abs_diff` without the need for any fallible conversions what so ever. (the suggestion stems from the fact that `i64` encodes just like `u64` over the network, i.e. if we ever change this to `u64` encoding we will still be backward compatible. will have to double check this fact still :) ).

Reply:
> still not sure how the negative side is encoded

Theoretically, it should be fine on encoding but should panic on decoding, which is the main purpose on the decoding side (as negative ones are invalid).